File: yoga-imessager/src/main/java/com/yoga/imessager/user/repo/UserRepository.java
package com.yoga.imessager.user.repo;

import com.yoga.imessager.user.model.User;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

@Repository(value = "imUserRepository")
public interface UserRepository extends JpaRepository<User, Long> {

    User findFirstByTenantIdAndId(long tenantId, long id);
    Page<User> findByTenantIdAndNickname(long tenantId, String nickname, Pageable pageable);
}

File: yoga-wechat/src/main/kotlin/com/yoga/wechat/users/UserRepository.kt
package com.yoga.wechat.users

import com.yoga.core.repository.BaseRepository
import org.springframework.data.jpa.repository.Modifying
import org.springframework.data.jpa.repository.Query
import org.springframework.stereotype.Component
import org.springframework.transaction.annotation.Transactional

@Component(value = "weixinUserRepository")
interface UserRepository: BaseRepository<User, Long> {

    @Transactional
    @Modifying
    @Query(value = "delete from wx_users where tenant_id = ?1 and account_id = ?2 and batch_index != ?3", nativeQuery = true)
    fun deleteByTenantIdAndAccountIdAndNotBatchIndex(tenantId: Long, accountId: Long, batchIndex: Long)

    fun findOneByTenantIdAndAccountIdAndOpenId(tenantId: Long, accountId: Long, openId: String): User?
    fun findOneByAccountIdAndOpenId(accountId: Long, openId: String): User?
    fun findByTenantIdAndAccountIdAndOpenIdIn(tenantId: Long, accountId: Long, openIds: Array<String>): List<User>
}
